sycl/doc/extensions/proposed/sycl_ext_oneapi_root_group.asciidoc

```diff
@@ -307,6 +340,65 @@ work-item is the leader of the root-group, and `false` for all other work-items
 in the root-group. The leader of the root-group is guaranteed to be the
 work-item for which `get_local_id()` returns 0.
 
+[source,c++]
+----
+template <execution_scope Scope>
+std::enable_if_t<(Scope == execution_scope::work_item ||
+                  Scope == execution_scope::work_group),
+                 id<Dimensions>>
+get_id() const;
+----
+_Returns_: An id representing the index of the current group or item at `Scope`
+hierarchy level within the root_group object.
+
+[source,c++]
+----
+template <execution_scope Scope>
+std::enable_if_t<Scope == execution_scope::sub_group, id<1>> get_id() const;
+----
+_Returns_: An id representing the index of the current sub_group within the
+root_group object.
+
+[source,c++]
+----
+template <execution_scope Scope>
+size_t get_linear_id() const;
+----
+_Returns_: A linearized number of the current group or item at `Scope` hierarchy
+level within the root_group object.
+
+[source,c++]
+----
+template <execution_scope Scope>
+std::enable_if_t<(Scope == execution_scope::work_item ||
+                  Scope == execution_scope::work_group),
+                 range<Dimensions>>
+get_range() const;
+----
+_Returns_: A range representing the number of groups or items of `Scope`
+hierarchy level within the root_group object.
+
+[source,c++]
+----
+template <execution_scope Scope>
+std::enable_if_t<Scope == execution_scope::sub_group, range<1>>
+get_range() const;
+----
+_Returns_: A range representing the number of sub_group's within the root_group
+object.
+
+[source,c++]
+----
+template <execution_scope Scope>
+size_t get_linear_range() const;
+----
+_Returns_: The number of groups or items of `Scope` hierarchy level within the
+root_group object.
+
+NOTE: Functions accepting `Scope` template parameter accept `Scope` of a lower
+hierarchy than `execution_scope::root_group`. Otherwise it results in a
+compilation error.
```

Comment:
This is currently formatted as a non-normative note, but the restriction on `Scope` values is part of the specification.

I think we should document the restriction on each function.  This will lead to some duplication in the specification, but will be unambiguous.  Whether this restriction should be a "Constraint" or a "Mandate" depends on whether you intend for the functions to be available or just give error messages: see [here](https://eel.is/c++draft/structure.specifications#3).

Reply:
Added constraint with [2855637](https://github.com/intel/llvm/pull/12643/commits/285563706f85a22fc3bcd794089b1d681bd81b9a)